### PR TITLE
Add CONTEXT_DIR build parameter

### PR DESCRIPTION
**Pull request by pdmack**
The S2I build model allows users to specify a context directory that will be the relative build location from a root git repo location, which is fairly common. This change adds a CONTEXT_DIR build parameter (e.g., `oc new-app --context-dir=<...>`) for each of the six supported oshinko S2I build templates.

### CODE REVIEW

**Comment by pdmack** [comment]
There might be some question about adding this or instead instructing developers to go the `oc new-build ...` path to manage this setting.

**Comment by pdmack** [comment]
Upstream closed https://github.com/openshift/origin/issues/14960 with a "no, go away" solution.
So PR still applies IMHO.

File: scala/scalabuild.json [hunk 1]
```diff
@@ -67,7 +71,9 @@
                "git": {
                   "uri": "${GIT_URI}",
                   "ref": "${GIT_REF}"
-               }
+               },
+               "contextDir": "${CONTEXT_DIR}"
+             }
```
Comment (tmckayus) [inline scala/scalabuild.json]: extra square bracket here

**Comment by tmckayus** [comment]
Kept forgetting about this.  Works for me, except for extra set brace in scalabuild.json.

Probably should add some s2i template tests for this functionality, too, but that can be another PR

**Pull request merged**